src/renderer/widgets/WalletDetails/ui/WalletActions/GeneralWalletActions.tsx

```diff
@@ -0,0 +1,48 @@
+import { DropdownIconButton } from '@shared/ui';
+import { Wallet } from '@shared/core';
+import { useI18n } from '@app/providers';
+import { useToggle } from '@shared/lib/hooks';
+import { RenameWalletModal } from './actions/RenameWalletModal';
+import { DropdownIconButtonOption } from '@shared/ui/Dropdowns/common/types';
+import { IconNames } from '@shared/ui/Icon/data';
+
+type Props = {
+  wallet: Wallet;
+  extraActions?: DropdownIconButtonOption[];
+};
+
+export const GeneralWalletActions = ({ wallet, extraActions }: Props) => {
+  const { t } = useI18n();
+  const [isRenameModalOpen, toggleIsRenameModalOpen] = useToggle();
+
+  const generalActions = [
+    {
+      icon: 'rename' as IconNames,
+      title: t('walletDetails.common.renameButton'),
+      onClick: toggleIsRenameModalOpen,
+    },
+    // {
+    //   icon: 'forget',
+    //   title: t('walletDetails.common.forgetButton'),
+    //   onClick: () => {},
+    // },
+  ];
+
+  const options = extraActions ? [...generalActions, ...extraActions] : generalActions;
+
+  return (
+    <>
+      <DropdownIconButton name="more">
+        <DropdownIconButton.Items>
+          {options.map((option) => (
+            <DropdownIconButton.Item key={option.icon}>
+              <DropdownIconButton.Option option={option} />
+            </DropdownIconButton.Item>
+          ))}
+        </DropdownIconButton.Items>
+      </DropdownIconButton>
+      {/*<DropdownIconButton className="m-1.5" name="more" options={options} optionsClassName="right-0" />*/}
+      <RenameWalletModal wallet={wallet} isOpen={isRenameModalOpen} onClose={toggleIsRenameModalOpen} />
```

Comment:
I don't like this approach of unification
`Rename` & `Forget` modals will stay here but `extraActions` will always stay outside of this component, so I should look here and there to understand what actions I'm able to do
If you put all available actions in dedicated `wallet details modal` then you will see everything at once and have the whole picture 
The ideal solution to remove duplication is to use modal manager 😊 which requires replacement for headlessui modal